pkg/git/fetch.go

```diff
@@ -94,21 +120,103 @@ func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
 	return nil
 }
 
-func resolveRevision(repository *git2go.Repository, gitRevision string) (*git2go.Oid, error) {
-	ref, err := repository.References.Dwim(gitRevision)
+func getHttpsTransport() (transport.Transport, error) {
+	if httpsProxy, exists := os.LookupEnv("HTTPS_PROXY"); exists {
+		parsedUrl, err := url.Parse(httpsProxy)
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing HTTPS_PROXY url")
+		}
+		proxyClient := &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyURL(parsedUrl),
+			},
+			Timeout: 15 * time.Second,
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		}
+		return githttp.NewClient(proxyClient), nil
+	} else {
+		return githttp.DefaultClient, nil
+	}
+}
+
+func copyFile(dstPath string, srcPath string) error {
+	srcFile, err := os.Open(srcPath)
 	if err != nil {
-		return resolveCommit(gitRevision)
+		return err
+	}
+	defer srcFile.Close()
+
+	dstFile, err := os.Create(dstPath)
+	if err != nil {
+		return err
 	}
+	defer dstFile.Close()
 
-	return ref.Target(), nil
+	_, err = io.Copy(dstFile, srcFile)
+	if err != nil {
+		return err
+	}
+
+	srcInfo, err := os.Stat(srcPath)
+	if err != nil {
+		return err
+	}
+
+	return os.Chmod(dstPath, srcInfo.Mode())
 }
 
-func resolveCommit(gitRevision string) (*git2go.Oid, error) {
-	oid, err := git2go.NewOid(gitRevision)
+// recursively copy the source directory to the destination directory
+func copyDir(dstDir string, srcDir string) error {
+	srcInfo, err := os.Stat(srcDir)
+	if err != nil {
+		return err
+	}
+
+	err = os.MkdirAll(dstDir, srcInfo.Mode())
+	if err != nil {
+		return err
+	}
+
+	entries, err := os.ReadDir(srcDir)
 	if err != nil {
-		return nil, errors.Errorf("could not find reference: %s", gitRevision) //invalid oid
+		return err
+	}
+
+	fileInfos, err := getFileInfos(entries)
+	if err != nil {
+		return err
+	}
+
+	for _, fileInfo := range fileInfos {
+		srcPath := path.Join(srcDir, fileInfo.Name())
+		dstPath := path.Join(dstDir, fileInfo.Name())
+
+		if fileInfo.IsDir() {
+			if err = copyDir(dstPath, srcPath); err != nil {
+				return err
+			}
+		} else {
+			if err = copyFile(dstPath, srcPath); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func getFileInfos(entries []os.DirEntry) ([]fs.FileInfo, error) {
+	fileInfos := make([]fs.FileInfo, 0, len(entries))
+	for _, entry := range entries {
+		info, err := entry.Info()
+		if err != nil {
+			return nil, err
+		}
+		fileInfos = append(fileInfos, info)
 	}
-	return oid, nil
+	return fileInfos, nil
 }
```

Comment:
Instead of fetching to a temporary directory and copying to `dir`, we now fetch directly to `dir`. Also added tests to verify that symbolic links are preserved and executable bits are preserved.